File: app/src/main/java/com/jameschamberlain/footballteamtracker/fixtures/Score.java
package com.jameschamberlain.footballteamtracker.fixtures;

import androidx.annotation.Nullable;

import java.io.Serializable;

public class Score implements Serializable {

    private int home;
    private int away;

    public Score() {
        this.home = -1;
        this.away = -1;
    }

    public Score(int home, int away) {
        this.home = home;
        this.away = away;
    }

    public int getHome() {
        return home;
    }

    public void setHome(int home) {
        this.home = home;
    }

    public int getAway() {
        return away;
    }

    public void setAway(int away) {
        this.away = away;
    }

    @Override
    public String toString() {
        if (home == -1 && away == -1) {
            return "- : -";
        } else {
            return home + " : " + away;
        }
    }

    @Override
    public boolean equals(@Nullable Object obj) {
        if (obj == this) return true;

        if (!(obj instanceof Score)) return false;

        Score s = (Score) obj;

        return home == s.home && away == s.away;
    }
}

File: app/src/main/java/com/jameschamberlain/footballteamtracker/objects/Score.kt
package com.jameschamberlain.footballteamtracker.objects

import java.io.Serializable

class Score : Serializable {
    var home: Int
    var away: Int

    internal constructor() {
        home = -1
        away = -1
    }

    constructor(home: Int, away: Int) {
        this.home = home
        this.away = away
    }

    override fun toString(): String {
        return if (home == -1 && away == -1) {
            "- : -"
        } else {
            "$home : $away"
        }
    }

    override fun equals(other: Any?): Boolean {
        if (other === this) return true
        if (other !is Score) return false
        return home == other.home && away == other.away
    }

    override fun hashCode(): Int {
        var result = home
        result = 31 * result + away
        return result
    }
}